src/Symfony/Component/Form/Util/OrderedHashMap.php

```diff
@@ -144,7 +144,7 @@ public function offsetSet($key, $value)
      */
     public function offsetUnset($key)
     {
-        if (false !== ($position = array_search($key, $this->orderedKeys))) {
+        if (false !== ($position = array_search((string) $key, $this->orderedKeys))) {
```

Comment:
Wouldn't it have been enough to pass `true` as the third argument here?

Reply:
@xabbuh https://3v4l.org/vBboa
When unsetting a numeric string offset like `unset($map['1'])`, `$key` equals to integer `1`. So it's needed to treat these numeric strings the same way as integers. Hence the cast. And since the `orderedKeys` ale all strings, passing `true` would result in the same.